automation/scripts/clean_pipelines/clean_pipelines.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import typing
+from time import sleep
+
+import pandas as pd
+import pytz
+from dateutil import parser
+from kfp import Client
+
+import mlrun
+import mlrun_pipelines
+from mlrun_pipelines.models import PipelineRun
+
+# Interval for logging deletion progress
+DELETION_LOG_INTERVAL = 100
+
+
+def delete_project_old_pipelines(
+    context: mlrun.MLClientCtx,
+    project_name: str,
+    end_date: str,
+    start_date: str = "",
+    dry_run: bool = True,
```

Comment:
readme says the default is false
```suggestion
    dry_run: bool = False,
```